src/Internal/LoggerContext.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Class LoggerContext
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal;
+
+/**
+ * Logger Context class.
+ */
+class LoggerContext {
+	/**
+	 * Request identifier.
+	 *
+	 * @var string
+	 */
+	private $request_id;
+
+	/**
+	 * Context data.
+	 *
+	 * @var array<string, string>
+	 */
+	private $context = [];
+
+	/**
+	 * Whether the hooks have been set.
+	 *
+	 * @var bool
+	 */
+	private $hooks_set = false;
+
+	/**
+	 * Initialises the logger context.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->request_id = uniqid();
+		$this->context    = [];
+
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Sets a context value.
+	 *
+	 * @param string                     $key   The key to set.
+	 * @param string|int|float|bool|null $value The value to set. Null removes value.
+	 *
+	 * @return void
+	 */
+	public function set_value( $key, $value ) {
+		if ( null === $value && array_key_exists( $key, $this->context ) ) {
+			unset( $this->context[ $key ] );
+		} else {
+			$this->context[ $key ] = (string) $value;
+		}
+	}
+
+	/**
+	 * Filter the log entry to include the request ID and context.
+	 *
+	 * @param string $entry   Log entry.
+	 * @param array  $context Log entry context.
+	 * @return string
+	 */
+	public function filter_log_entry( $entry, $context ): string {
+		$entry_context = is_array( $context ) && array_key_exists( 'context', $context )
+			? $context['context']
+			: [];
+		if ( ! array_key_exists( 'source', $entry_context ) || 'woopayments' !== $entry_context['source'] ) {
```

Comment:
Let's use `Logger::LOG_FILENAME` instead of `woopayments` here:
```suggestion
		if ( ! array_key_exists( 'source', $entry_context ) || Logger::LOG_FILENAME !== $entry_context['source'] ) {
```

Overall, I was about to suggest significantly simplifying this set of conditions, but I realized that you might want to use the `$entry_context` to add to the log. Is that the case? Because otherwise, the third parameter for `Logger->log()` is useless, as it does not automatically appear in the output.

Reply:
I think that we can eventually drop the changes to pass `context` to the Logger, as it won't make sense for WooPayments. On an interesting note, WooCommerce logger appends context information, if it is present, to the end of the log entry.

Reply:
> WooCommerce logger appends context information, if it is present, to the end of the log entry.

I tested this briefly, and the context that I provided did not appear in the log file. I'll look into how Woo core does it.

Reply:
It is not in the WooPayments log, because entries are processed by the new filter. But if you disable filter, you will see it.